### PR TITLE
First draft for the inventory turnover table :)

### PR DESCRIPTION
**Pull request by hmmoek**

### CODE REVIEW

File: models/product_explorer/inventory_turnover.sql [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        a.order_date,
+        a.brand_name,
+        a.category_description,
+        b.stock_qty,
+        a.year,
+        case 
+            when month=1  or month=2  or month=3  then 1
+            when month=4  or month=5  or month=6  then 2
+            when month=7  or month=8  or month=9  then 3
+            when month=10 or month=11 or month=12 then 4
```
Comment (aTiselius) [inline models/product_explorer/inventory_turnover.sql:31]: Date ( querter, date value)

Comment (aTiselius) [reply models/product_explorer/inventory_turnover.sql:31]: EXTRACT(QUARTERFROM datevalue)

Comment (martenlindblad) [reply models/product_explorer/inventory_turnover.sql:31]: yes, and move it to line 10 or something :)

File: models/product_explorer/inventory_turnover.sql [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        end as quarter
+    from wp_sales as a
+    inner join stock_value as b
+    on date(a.order_date)=b.stock_date and a.product_sku=b.product_sku
```
Comment (martenlindblad) [inline models/product_explorer/inventory_turnover.sql:35]: move the cast to the cte wp_sales comes from.

File: models/product_explorer/inventory_turnover.sql [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+final as (
+    select 
+        product_sku,
+        order_date,
+        brand_name,
+        category_description,
+        case 
+            when (3*avg_last_3_q+part_past_q*avg_qty_current_q)=0 then null
+            else 4*qty_sold_q/(3*avg_last_3_q+part_past_q*avg_qty_current_q)*(3+part_past_q) --where the part after "qty_sold_q"/ is average items in stock last 4 quarters
+        end as turnover_rate,
```
Comment (martenlindblad) [inline models/product_explorer/inventory_turnover.sql:87]: Hänger inte riktigt med på vad den räknar ut? Finns det någon formel du kan hänvisa till?

File: models/product_explorer/inventory_turnover.sql [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        avg(avg_stock_qty_q) over (partition by product_sku order by year_and_q rows between 3 preceding and 1 preceding) as avg_last_3_q --so not including the current q
+    from calc_avg_stock_q
+),
+
+merged as (
+    select 
+        a.*,
+        ifnull(b.avg_last_3_q, 0) as avg_last_3_q,
+        case 
+            when a.quarter=1 then date_diff( date(order_date),date(concat(cast(a.year as string),'-01-01') ),day) / date_diff( date(concat(cast(a.year as string),'-03-31')),date(concat(cast(a.year as string),'-01-01')),day)
```
Comment (aTiselius) [inline models/product_explorer/inventory_turnover.sql:68]: {%- set metric = ' date(order_date),date(concat(cast(a.year as string' -%}
when a.quarter=1 then date_diff({{metric}}),'-01-01') ),day)
:) ? om du gillar det

File: models/product_explorer/inventory_turnover.sql [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+merged as (
+    select 
+        a.*,
+        ifnull(b.avg_last_3_q, 0) as avg_last_3_q,
+        case 
+            when a.quarter=1 then date_diff( date(order_date),date(concat(cast(a.year as string),'-01-01') ),day) / date_diff( date(concat(cast(a.year as string),'-03-31')),date(concat(cast(a.year as string),'-01-01')),day) 
+            when a.quarter=2 then date_diff( date(order_date),date(concat(cast(a.year as string),'-04-01') ),day) / date_diff( date(concat(cast(a.year as string),'-07-31')),date(concat(cast(a.year as string),'-04-01')),day)
+            when a.quarter=3 then date_diff( date(order_date),date(concat(cast(a.year as string),'-07-01') ),day) / date_diff( date(concat(cast(a.year as string),'-09-30')),date(concat(cast(a.year as string),'-07-01')),day)
+            when a.quarter=4 then date_diff( date(order_date),date(concat(cast(a.year as string),'-10-01') ),day) / date_diff( date(concat(cast(a.year as string),'-12-31')),date(concat(cast(a.year as string),'-10-01')),day)
+        end as part_past_q --part of the days in the current quarter that have past already
```
Comment (martenlindblad) [inline models/product_explorer/inventory_turnover.sql:72]: denna rad blir samma resultat som hela case satsen. `date_diff(order_date, date_trunc(order_date, quarter), day) / date_diff(date_sub(date_trunc(date_add(order_date, INTERVAL 1 quarter), quarter), INTERVAL 1 day), date_trunc(order_date, quarter), day)`

**Pull request merged**